packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx

```diff
@@ -68,22 +67,38 @@ const Zone = ({
       contentType: getContentTypeForTracking(window.location.pathname),
     });
   };
-  const enterSelectMode = () => {
-    setSelectModalData({ isOpen: true });
-  };
 
   const closeEditModal = () => {
     setEditModalData({ isOpen: false });
   };
-  const closeSelectModal = () => setSelectModalData({ isOpen: false });
 
   const onSelectFieldType = (widgetTypeName) => {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
     setNewFieldData({ widgetTypeName, fields });
-    setSelectModalData({ isOpen: false });
   };
 
   const onCancelNewField = () => setNewFieldData(null);
+  const addFieldDropdown = (
+    <AddFieldDropdown
+      disabled={newFieldData !== null}
+      onSelectField={onSelectFieldType}
+      fields={
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
```

Comment:
❓ What's the problem here? Can't it be solved with an assert?

Reply:
I don't even know nor try x)
It's old JS code from this file that I try not to touch, all JS files are so ugly with all of that. I'm focusing myself really on extracting code a little by little and prevent touching anything there.

I could try to prevent using all of this also, of course. WDYT?

Reply:
If it's legacy that's ok! we don't have to waste time on it 👍